bin/BuildPackages.sh

```diff
@@ -21,55 +50,82 @@ set -e
 # an idea what to do for a complete installation of GAP.
 
 
-if [ $# -eq 0 ]
-  then
-    echo "Assuming default GAP location: ../.."
-    GAPDIR=../..
-  else
-    echo "Using GAP location: $1"
-    GAPDIR="$1"
+# Is someone trying to run us from inside the 'bin' directory?
+if [[ -f gapicon.bmp ]]
+then
+  error "This script must be run from inside the pkg directory" \
+        "Type: cd ../pkg; ../bin/BuildPackages.sh"
 fi
 
-# Is someone trying to run us from inside the 'bin' directory?
-if [ -f gapicon.bmp ]
-  then
-    echo "This script must be run from inside the pkg directory"
-    echo "Type: cd ../pkg; ../bin/BuildPackages.sh"
-    exit 1
+# CURDIR
+CURDIR="$(pwd)"
+
+# GAPDIR
+if [[ "$#" -ge 1 ]]
+then
+  case "$1" in
+    --with-gaproot=*)
+      GAPDIR="$(sed 's|--with-gaproot=||' <<< $1)"
+      shift
+    ;;
+
+    *)
+      cd ..
+      GAPDIR="$(pwd)"
+      cd "$CURDIR"
+    ;;
```

Comment:
Aha, it seems that (reading on) you are simply changing the semantics of this script. I guess I don't terribly mind, but it's not super nice either.. Hrm.

You could make your intent clearer, though, and at the same time reduce code duplication, by restructung the code to look like this:
```sh
# GAPDIR
GAPDIR=
if [[ "$#" -ge 1 ]]
then
  case "$1" in
    --with-gaproot=*)
      GAPDIR=$(sed 's|--with-gaproot=||' <<< "$1")
      shift
    ;;
fi
if [[ -z "$GAPDIR" ]]
then
  pushd ..
  GAPDIR="$(pwd)"
  popd
fi
notice "Using GAP location: $GAPDIR"
```